### ISSUE TITLE
installing talk with talk.sh on nextcloud 18 fails

### ISSUE THREAD

**Issue by u-hafner** [nextcloud/vm#1050]
I tried to install talk to nextcloud 18 a day after upgrading from 17 to 18 yesterday (update with /var/srcipts/update.sh). used /var/scripts/talk.sh
It failed with the error messages in cli output:
Errors were encountered while processing:
 docker-ce
E: Sub-process /usr/bin/dpkg returned an error code (1)
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
install_if_not coturn failed

Steps to reproduce
0. Installation was an installation of nextcloud 17 with the script provided by nextcloud/vm 3 month ago. No problems.
1. Updated nextcloud 17 to 18 with update.sh yesterday
2. try to install talk with /var/scripts/talk.sh today

## Expected behaviour
Wanted to install talk to nextcloud 18

## Actual behaviour
got the error message: install_if_not coturn failed
preceded by: Errors were encountered while processing:
docker-ce

## Server configuration

**Nextcloud server version:** 18.0.0.10

**Server OS** Ubuntu 18.04 Container running on Proxmox 5.4

**How did you install the VM?** don't know

## Network
**Do you use DHCP?**
yes
**Is port 80 and/or 443 open?**
yes

## Logs / Screenshots
update.log only contains:
NEXTCLOUD UPDATE success-20200121

nextcloud.log does not contain error messages. only ldap debug messages.

#### Installation log (command output) CLI Output
[https://0bin.net/paste/tAlewO171KKvpSMN#YEVw70DPv10cOmdV909227mD3Cg0M1cHJNC9ptC4ax7](https://0bin.net/paste/tAlewO171KKvpSMN#YEVw70DPv10cOmdV909227mD3Cg0M1cHJNC9ptC4ax7)

**Comment by enoch85** [comment]
This seems to be an issue with the docker package in Ubuntu, not the script.

> It failed with the error messages in cli output:
Errors were encountered while processing:
docker-ce
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by u-hafner** [comment]
oh. ok. Thanks for the fast answer. Any hint for me where I can get more infos to look for?
I have no idea about docker. It was installed with this installationscript of nextcloud/vm

**Comment by enoch85** [comment]
Docker is only installed if needed. So did yo install Collabora, FullTextSearch or OnlyOffice?

**Comment by u-hafner** [comment]
FullTextSearch is installed. Sounded like something usefull ;-).
Guess I will dump my db and reinstall clean without that. Hope i can reapply my database.
Actually looking at the upgrade from 17 to 18 I just saw the docker-ce problems when I retried after applying a backup of the 17 version.
Otherwise I will always have that issue trying to update nextcloud already. I better clean this up before our nextcloud gets used heavily.

Thanks.

**Comment by enoch85** [comment]
You shouldn't have to restore DB. It's just a package issue. 

Try `sudo apt-hold docker*` and then try to run the update script again.

**Comment by u-hafner** [comment]
You meant: sudo apt-mark hold I hope. Didn't find the command you posted. That worked. Thanks. Now my upgrade of nextcloud had no warnings anymore and installing talk I got:
Installing spreed...
PHP Fatal error:  Cannot declare class OCA\Talk\Migration\Version2000Date20170707093535, because the name is already in use in /var/www/nextcloud/apps/spreed/lib/Migration/Version2000Date20170707093535.php on line 127
spreed enabled
Config value stun_servers for app spreed set to ["our-server:5349"]
Config value turn_servers for app spreed set to [{"server":"our-server:5349","secret":"the secret","protocols":"udp,tcp"}]

It says it installed successfully. 
I tested it and it worked so I assume that message did not mean any trouble for talk.

Thanks again.
